src/plenoptic/simulate/models/portilla_simoncelli.py

```diff
@@ -84,20 +84,19 @@ def __init__(
             tight_frame=False,
         )
         self.filterPyr = SteerablePyramidFreq(
-            self.pyr._lomasks[-1].shape[-2:], height=0, order=1,
+            getattr(self.pyr, f'_lomasks_scale_{n_scales-1}').shape[-2:],
+            height=0, order=1,
             tight_frame=False
         )
-        self.unoriented_band_pyrs = [
```

Comment:
I think you can use `nn.ModuleList` to keep `self.unoriented_band_pyrs` as a list but still have the desired device/dtype behavior.